sawmill-core/src/main/java/io/logz/sawmill/conditions/IpCompareCondition.java

```diff
@@ -0,0 +1,114 @@
+package io.logz.sawmill.conditions;
+
+import io.logz.sawmill.Condition;
+import io.logz.sawmill.Doc;
+import io.logz.sawmill.annotations.ConditionProvider;
+import io.logz.sawmill.parser.ConditionParser;
+import io.logz.sawmill.utilities.JsonUtils;
+
+import java.net.InetAddress; 
+import java.net.UnknownHostException; 
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+@ConditionProvider(type = "ipRange", factory = IpCompareCondition.Factory.class)
+public class IpCompareCondition implements Condition  {
+	
+	private String ipHigh;
+	private String ipLow;
+	private String field;
+	
+	public IpCompareCondition(String field, String ipLow, String ipHigh) {
+		this.field = requireNonNull(field);
+		this.ipHigh = requireNonNull(ipHigh);
+		this.ipLow = requireNonNull(ipLow);
+		
+	}
+	
+	public static long ipToLong(InetAddress ip) {
+        byte[] octets = ip.getAddress();
+        long result = 0;
+        for (byte octet : octets) {
+            result <<= 8;
+            result |= octet & 0xff;
+        }
+        return result;
+    }
+	
+    @Override
+    public boolean evaluate(Doc doc) {
+        
+	if (!doc.hasField(field, String.class)) return false;
```

Comment:
indentation seems off, please do `command+optional+L` on this whole file to arrange it